### ISSUE TITLE
Type info of expressions with parentheses is not mirrored in pyright output

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#1892]
Type info of expressions with parentheses is not mirrored in the pyright output, which can result in somewhat confusing output:

```py
reveal_type((["hi"] * 2)[0]) # Type of "["hi"] * 2[0]" is "str"
```

In this case, the expression enclosed in quotes would not be valid python code because of the missing parentheses.

Tested on pyright `1.1.141`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.142, which I just published. It will also be included in the next release of pylance.
